pylint/extensions/check_mccabe.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+"""Module to add McCabe checker class for pylint. """
+
+import ast
+
+from mccabe import McCabeChecker
+from pylint.checkers.base import BaseChecker
+from pylint.checkers.utils import check_messages
+from pylint.interfaces import HIGH, IAstroidChecker
+
+
+class McCabeMethodChecker(BaseChecker):
+    """Checks McCabe complexity cyclomatic threshold in methods and functions
+    to validate a too complex code.
+    """
+
+    __implements__ = IAstroidChecker
+    name = 'design'
+
+    msgs = {
+        'R1260': (
+            '%s is too complex. The McCabe rating is %d',
+            'too-complex',
+            'Used when a method or function is too complex based on '
+            'McCabe Complexity Cyclomatic'),
+    }
+    options = (
+        ('max-complexity', {
+            'default': 10,
+            'type': 'int',
+            'metavar': '<int>',
+            'help': 'McCabe complexity cyclomatic threshold',
+        }),
+    )
+
+    @staticmethod
+    def _get_ast_tree(node):
+        """Get a compile python tree of nodes from code"""
+        # TODO: Check if astroid build a similar tree
+        code = node.as_string()
+        tree = compile(code, '', "exec", ast.PyCF_ONLY_AST)
```

Comment:
Why do you this? We already have the tree built for us, available through visit_module. Also, if McCabeChecker works only with a builtin ast module, instead of the one provided by astroid, then we should implement the complexity metric ourselves.